libgadget/blackhole.c

```diff
@@ -231,6 +323,28 @@ add_injected_BH_energy(double unew, double injected_BH_energy, double mass)
     return unew;
 }
 
+/* check if two BHs are gravitationally bounded, input dv, da, dx in code unit */
+/* same as Bellovary2011, Tremmel2017 */
+static int
+check_grav_bound(double dx[3], double dv[3], double da[3])
+{
+    int j;
+    double KE = 0;
+    double PE = 0;
+
+    for(j = 0; j < 3; j++){
+        KE += 0.5 * pow(dv[j], 2);
+        PE += da[j] * dx[j];
+    }
+    
+    KE /= (All.cf.a*All.cf.a); /* convert to proper velocity */
+    PE /= All.cf.a; /* convert to proper unit */
+    
```

Comment:
Shall we briefly review the h and a_scale factor in dx, dv, and da here?

From my understanding:
dx = v * drift_fac = v * a^{-2} dt             // drift_integ = da/(H(a)*a^3) = dt/a^2
dv = acc * kick_fac = acc * a^{-1} dt     // kick_integ = da/(H(a)*a^2) = dt/a

for a_scale factor,
=> dx ~ a (by '~', I mean dx * a = dx_proper), dv ~ a^{-1},  acc ~ a^{-2}
Therefore  KE /= a^2, PE /= a  converts to proper unit.

for h factor,
=> dx ~ h^{-1},  dt ~ h^{-1},  dv ~ h^{0}, d_acc ~ h^{1}
Therefore, da * dx has h factor canceled out, there's no h factor in both KE and PE.